### PR TITLE
Fixing QiitaClient

### PR DESCRIPTION
**Pull request by josenavas**
Depends on #1794 so review/merge that one first.

The QiitaClient is improved to reduce workload on the plugin developer.

### CODE REVIEW

File: qiita_plugins/qiita_client/qiita_client/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+
+class QiitaClientError(Exception):
```
Comment (antgonza) [inline qiita_plugins/qiita_client/qiita_client/exceptions.py]: What's the plan for these?

**Comment by antgonza** [comment]
One question, then 👍 when test pass.

File: qiita_plugins/qiita_client/qiita_client/exceptions.py (hunk 1 above)
Comment (josenavas) [inline qiita_plugins/qiita_client/qiita_client/exceptions.py]: They're used in the Qiita client and I think there are 3 different errors that the developer using the QiitaClient can decide what to do (either propagate the error or try something else). For example, this allows me to code a different behavior in the `_heartbeat` function below.

Comment (antgonza) [inline qiita_plugins/qiita_client/qiita_client/exceptions.py]: K, thanks.

**Comment by josenavas** [comment]
@ElDeveloper available for a quick review in here?

File: qiita_plugins/qiita_client/qiita_client/qiita_client.py [hunk 2]
```diff
@@ -24,16 +27,30 @@ def heartbeat(qclient, url):
...
+    retries = 2
+    while not JOB_COMPLETED and retries > 0:
+        try:
+            qclient.post(url, data='')
+            retries = 2
+        except requests.ConnectionError:
+            # This error occurs when the Qiita server is not reachable. This
+            # may occur when we are updating the server, and we don't want
+            # the job to fail. In this case, we wait for 5 min and try again
+            time.sleep(300)
```
Comment (ElDeveloper) [inline qiita_plugins/qiita_client/qiita_client/qiita_client.py]: Would it be worth mentioning this in the docstring?

File: qiita_plugins/qiita_client/qiita_client/qiita_client.py [hunk 3]
```diff
@@ -343,15 +384,23 @@ def update_job_step(self, job_id, new_step):
...
         """Stops the heartbeat thread and send the job results to the server
 
         Parameters
         ----------
         job_id : str
             The job id
-        payload : dict
-            The job's results
+        success : bool
+            Whether if the job completed successfully or not
```
Comment (ElDeveloper) [inline qiita_plugins/qiita_client/qiita_client/qiita_client.py]: You can remove the `if` from this sentence.

**Comment by ElDeveloper** [comment]
👍

File: qiita_plugins/qiita_client/qiita_client/qiita_client.py (hunk 2 above)
Comment (josenavas) [inline qiita_plugins/qiita_client/qiita_client/qiita_client.py]: Done

File: qiita_plugins/qiita_client/qiita_client/qiita_client.py (hunk 3 above)
Comment (josenavas) [inline qiita_plugins/qiita_client/qiita_client/qiita_client.py]: Done

**Pull request merged**